app/src/editor/dashboard/index.jsx

```diff
@@ -210,9 +211,19 @@ const DashboardEdit = withRouter(class DashboardEdit extends Component {
     }
 })
 
-const DashboardLoader = withRouter(withErrorBoundary(ErrorComponentView)(class DashboardLoader extends React.PureComponent {
+const ErrorComponent = ({ error, ...props }) => {
+    if (error instanceof ResourceNotFoundError) {
+        throw error
+    }
+    return <ErrorComponentView {...props} error={error} />
+}
+
+const DashboardLoader = withRouter(withErrorBoundary(ErrorComponent)(class DashboardLoader extends React.PureComponent {
     static contextType = UndoContext
-    state = { isLoading: false }
+    state = {
+        isLoading: false,
+        error: null,
```

Comment:
`DashboardLoader` is not a funcional component. Didn't wanna refactor it so used `this.state` instead of `useFailure` hook.